concordium-sdk/src/main/java/com/concordium/sdk/transactions/smartcontracts/WasmModule.java

```diff
@@ -0,0 +1,70 @@
+package com.concordium.sdk.transactions.smartcontracts;
+
+import com.concordium.sdk.crypto.SHA256;
+import com.concordium.sdk.transactions.TransactionType;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.val;
+import org.apache.commons.codec.binary.Hex;
+
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Objects;
+
+@Getter
+public class WasmModule {
+
+    private final WasmModuleVersion version;
+    private final WasmModuleSource moduleSource;
+
+    @Builder
+    public WasmModule(WasmModuleVersion version, WasmModuleSource moduleSource) {
+        if (Objects.isNull(moduleSource)) {
+            throw new IllegalArgumentException("WasmModuleSource cannot be null");
+        }
+        this.version = version;
+        this.moduleSource = moduleSource;
+    }
+
+    public static WasmModule from(byte[] fileBuffer, int version) {
+        WasmModuleSource moduleSource;
+        WasmModuleVersion moduleVersion;
+        fileBuffer = Arrays.copyOfRange(fileBuffer, 8, fileBuffer.length);
+
+        if (version == 1) {
+            moduleSource = WasmModuleSourceV1.from(fileBuffer);
+            moduleVersion = WasmModuleVersion.V1;
+        }
+        else {
+            moduleSource = WasmModuleSourceV0.from(fileBuffer);
+            moduleVersion = WasmModuleVersion.V0;
+        }
+        return new WasmModule(moduleVersion, moduleSource);
+    }
+
+    /**
+     * Get the identifier of the WasmModule.
+     * The identifier is a SHA256 hash of the raw module bytes.
+     * @return the identifier of the Module.
+     */
+    public String getIdentifier() {
+        return Hex.encodeHexString(SHA256.hash(this.moduleSource.getBytes()));
+    }
+
+    /**
+     * Get the raw serialized bytes of the concrete {@link WasmModule}.
+     * This can currently either be {@link WasmModuleSourceV0} or {@link WasmModuleSourceV1}
+     *
+     * @return bytes
+     */
+
+    public byte[] getBytes() {
```

Comment:
Does this even work when the module is constructed via `cargo concordium`? 
`cargo concordium` embeds the version with the outputted `.wasm` file. The sdk should be able to handle this in some nice way. Maybe the provided `WasmModuleVersion` should be wrapped in an `Optional` indicating whether the `moduleSourceBytes` already contains the version tag or not.

Reply:
handled by function overloading. 
creator function now has option of just using a `byte[]` as module source Or a `byte[]` with `Version` depending on how the module was compiled.